### PR TITLE
Retrieve, decrypt and persist take-home-pay.

### PR DESCRIPTION
**Pull request by DanielChicot**

### CODE REVIEW

**Review by steveburton4 — DISMISSED** [review]

**Review by connoravo — DISMISSED** [review]

File: src/replayer_mismatch/crypto.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    decrypted_key_response = kms_client.decrypt(CiphertextBlob=cipher)
+    return decrypted_key_response["Plaintext"]
+
+
+def decrypted_take_home_pay(decrypted_key, encrypted_take_home_pay):
+    iv = encrypted_take_home_pay[:16]
+    encrypted = encrypted_take_home_pay[16:]
+    aes = AES.new(decrypted_key, AES.MODE_GCM, nonce=base64.urlsafe_b64decode(iv))
+    raw = base64.urlsafe_b64decode(encrypted)
+    return aes.decrypt_and_verify(raw[:-16], raw[-16:]).decode("ASCII")
```
Comment (MattToczek) [inline src/replayer_mismatch/crypto.py:16]: Is there a reason that you are using `[-16:]` and `[:-16]` on line 16 and `[16:]` and `[:16]` on line 12 and 13? Would be easier to read if they were the same format, but I can't work out if you are just splitting them in half, or there is another reason.

Comment (DanielChicot) [reply src/replayer_mismatch/crypto.py:16]: In the first one (that uses positive 16) I want to split on the 16th character, on the second one (the minus 16) I want to split on the 16th character from the end.

**Review by steveburton4 — APPROVED** [review]

File: src/replayer_mismatch/crypto.py (hunk 1 above)
Comment (MattToczek) [reply src/replayer_mismatch/crypto.py:16]: Ye, I understand what the functionality of the code is, I think my question (albeit badly worded) was more "is the string 32 chars long and you are splitting it down the middle or, are you taking a 16 char subset of a non 32 char string?"

**Review by MattToczek — APPROVED** [review]

File: src/replayer_mismatch/crypto.py (hunk 1 above)
Comment (DanielChicot) [reply src/replayer_mismatch/crypto.py:16]: Yep the latter, both strings are over 32 characters.

**Pull request merged**